lib/bandit/headers.ex

```diff
@@ -89,4 +83,18 @@ defmodule Bandit.Headers do
   end
 
   defp parse_integer(rest, total), do: {total, rest}
+
+  def add_content_length(headers, length, status) do
+    headers = Enum.reject(headers, &(elem(&1, 0) == "content-length"))
```

Comment:
Maybe

```suggestion
    headers = List.keydelete(headers, "content-length", 0)
```

Reply:
Alas, I want to delete *all* possible occurrences of content-length in the input header list, and `keydelete/3` only deletes the first.

In general I find the `List` functions inherited from `:list` to be super awkward to use (like, for example, `keyreplace` will only replace, not upsert). They're an oddball interface that I've never found the right place to make heavy use of.